src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -219,10 +219,12 @@ BlockIO InterpreterCreateQuery::createDatabase(ASTCreateQuery & create)
     else
     {
         bool is_on_cluster = getContext()->getClientInfo().query_kind == ClientInfo::QueryKind::SECONDARY_QUERY;
-        if (create.uuid != UUIDHelpers::Nil && !is_on_cluster)
+        if (create.uuid != UUIDHelpers::Nil && !is_on_cluster && !internal)
```

Comment:
Isn't `is_on_cluster` same as `internal`?

Reply:
Not always. `RestorerFromBackup` calls [InterpreterCreateQuery::execute()](https://github.com/ClickHouse/ClickHouse/blob/886a6e6e86f750c2d53d1f2dbdddf76b74061d86/src/Databases/DatabasesCommon.cpp#L368) with `internal` set to true. It's not ON CLUSTER queries.

And `is_on_cluster` is set when `CREATE TABLE ON CLUSTER` is executed, it has nothing to do with backups.

Reply:
I wrote some comment in the code.